src/test/java/org/ambraproject/rhino/service/impl/SolrIndexServiceTest.java

```diff
@@ -0,0 +1,222 @@
+ */
+@ContextConfiguration(classes = SolrIndexServiceTest.class)
+@Configuration
+public class SolrIndexServiceTest extends AbstractStubbingArticleTest {
+
+  private static final String EISSN = "EISSN-0001";
+
+  private static final String FILE_TYPE = "manuscript";
+
+  private static final String META_JOURNAL_NAME = "Meta journal name";
+
+  private static final String REPO_UUID = UUID.randomUUID().toString();
+
+  private ContentRepoService mockContentRepoService;
+
+  private RepoObjectMetadata mockRepoMetadata;
+
+  private ArticleCrudService mockArticleCrudService;
+
+  private MessageSender mockMessageSender;
+  
+  private SolrIndexService mockSolrIndexService;
+
+  private Doi expectedDoi;
+
+  private Article expectedArticle;
+
+  private Journal expectedJournal;
+
+  private ArticleIngestion expectedIngestion;
+
+  private ArticleIdentifier expectedArticleIdentifier;
+
+  private ArticleFile expectedArticleFile;
+
+  private ArticleItem expectedArticleItem;
+
+  /**
+   * Prepare test fixtures.
+   */
+  @BeforeMethod(alwaysRun = true)
+  protected void init() {
+    mockContentRepoService = buildMockContentRepoService(DESTINATION_BUCKET);
+    mockRepoMetadata = mock(RepoObjectMetadata.class);
+    mockArticleCrudService = applicationContext.getBean(ArticleCrudService.class);
+    mockMessageSender = applicationContext.getBean(MessageSender.class);
+    mockSolrIndexService = applicationContext.getBean(SolrIndexService.class);
+
+    expectedDoi = createStubArticleDoi();
+
+    expectedArticle = createStubArticle(1L /* article_id */, expectedDoi.getName());
+
+    expectedJournal = new Journal();
+    expectedJournal.setJournalKey(META_JOURNAL_NAME);
+    expectedJournal.seteIssn(EISSN);
+
+    expectedIngestion = createStubArticleIngestion(expectedArticle, INGESTION_NUMBER);
+    expectedIngestion.setArticle(expectedArticle);
+    expectedIngestion.setJournal(expectedJournal);
+
+    expectedArticleIdentifier = ArticleIdentifier.create(expectedDoi);
+
+    expectedArticleFile = new ArticleFile();
+    expectedArticleFile.setFileType(FILE_TYPE);
+    expectedArticleFile.setBucketName(DESTINATION_BUCKET);
+    expectedArticleFile.setCrepoKey(REPO_UUID);
+    expectedArticleFile.setCrepoUuid(REPO_UUID);
+
+    expectedArticleItem = new ArticleItem();
+    expectedArticleItem.setDoi(expectedDoi.getName());
+    expectedArticleItem.setIngestion(expectedIngestion);
+    expectedArticleItem.setFiles(ImmutableList.of(expectedArticleFile));
+  }
+
+  @Bean
+  public AssetCrudService assetCrudService() {
+    AssetCrudService mockAssetCrudService = spy(AssetCrudServiceImpl.class);
+    LOG.info("assetCrudService() * --> {}", mockAssetCrudService);
+    return mockAssetCrudService;
+  }
+
+  @Bean
+  public ArticleCrudService articleCrudService() {
+    final ArticleCrudService articleCrudService = spy(ArticleCrudServiceImpl.class);
+    return articleCrudService;
+  }
+
+  @Bean
+  protected MessageSender messageSender() {
```

Comment:
`AssetCrudService` and `articleCrudService` are already defined as beans in `AbstractStubbingArticleTest`, so no need to redeclare them here.

The `MessageSender` bean is also used by `SyndicationCrudServiceImplTest`. We should consider removing the code duplication and moving the bean declaration to  `AbstractStubbingArticleTest`

Reply:
For `AssetCrudService` and `ArticleCrudService`, they're **mock**ed in `AbstractStubbingArticleTest`; however, I wanted to **spy** on the beans which caused me to override the declarations.

Reply:
For `MessageSender` bean, I think for now we should keep it where it's needed.